ocaml-lsp-server/bin/main.ml

```diff
@@ -1,16 +1,61 @@
+open Stdune
+
 let () =
   Printexc.record_backtrace true;
   let version = ref false in
   let read_dot_merlin = ref false in
-  Arg.parse
+  let anon_args = ref [] in
+  let pipe = ref None in
+  let port = ref None in
+  let spec =
     [ ("--version", Arg.Set version, "print version")
+    ; ("--pipe", Arg.String (fun p -> pipe := Some p), "set pipe path")
+    ; ("--port", Arg.Int (fun p -> port := Some p), "set port")
     ; ( "--fallback-read-dot-merlin"
       , Arg.Set read_dot_merlin
       , "read Merlin config from .merlin files. The `dot-merlin-reader` \
          package must be installed" )
     ]
-    (fun _ -> raise (Arg.Bad "anonymous arguments are not accepted"))
-    "ocamllsp";
+  in
+  let usage = "ocamllsp [ stdio ]" in
+  Arg.parse spec (fun arg -> anon_args := arg :: !anon_args) usage;
+  let anon_args = List.rev !anon_args in
+  let channel =
+    match
+      match anon_args with
+      | [] -> Ocaml_lsp_server.Stdio
```

Comment:
we now only support stdio, but I think we should be checking here whether `!pipe` or `!port` is Some, right? was this omitted intentionally?

Reply:
Do you mean is None? Those arguments are useless with Stdio.

Reply:
I fixed the argument parsing a little.